Add turrets to A.I. and revamp A.I. core of several stations

OnsenCapy opened the pull request:
<!--
SPDX-FileCopyrightText: 2021 Pieter-Jan Briers <pieterjan.briers+git@gmail.com>
SPDX-FileCopyrightText: 2021 Swept <sweptwastaken@protonmail.com>
SPDX-FileCopyrightText: 2021 mirrorcult <lunarautomaton6@gmail.com>
SPDX-FileCopyrightText: 2022 AJCM-git <60196617+AJCM-git@users.noreply.github.com>
SPDX-FileCopyrightText: 2022 Kara <lunarautomaton6@gmail.com>
SPDX-FileCopyrightText: 2023 DrSmugleaf <DrSmugleaf@users.noreply.github.com>
SPDX-FileCopyrightText: 2023 Kevin Zheng <kevinz5000@gmail.com>
SPDX-FileCopyrightText: 2024 Vasilis <vasilis@pikachu.systems>
SPDX-FileCopyrightText: 2024 lzk <124214523+lzk228@users.noreply.github.com>
SPDX-FileCopyrightText: 2025 Aiden <28298836+Aidenkrz@users.noreply.github.com>

SPDX-License-Identifier: AGPL-3.0-or-later
-->

<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- NOTE: All code submitted to this repository is ALWAYS licensed under the AGPL-3.0-or-later license. 
The REUSE Specification headers or separate .license files indicate a secondary license (e.g., MPL or MIT) solely to facilitate 
integration for projects that do not use the AGPL license. This secondary license does not replace the fact that AGPL-3.0-or-later remains the primary and binding license. 
Uncomment and modify the following line if you wish to change the license from the default of AGPL.-->
<!--- LICENSE: AGPL -->
## About the PR
The following stations now have A.I. turrets and may have received changes to AI core:
- Delta

## Why / Balance
While the control panel for AI turrets is not yet added, the turrets are fully functional, just needing to be manually activated. I'm sure AI players would much rather have manual turrets than none at all.


## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc).
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X ] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [ X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

**Changelog**
:cl:
- tweak: Several stations have had their AI cores changed
- add: AI now has AI turrets, though the control panel has not yet been implemented. Manual activation of each turret is required by the AI.

SX-7 reviewed (approved):
don't do this in the future!!1 I mean the stacking of changes into one PR

ALSOOO why are you moving oasis? isn't it in the goobstation namespace already?

Gallagin commented:
Personally really don't like the change to my Leonid having the AI's outer asteroid area being extra fully sealed with walls now, it's not meant to be hard to get to the AI's building. 

Also you placed the blue floor decals on top of the warning stripes making that look weird. place the blue first then the hazard stripes
![2025-05-17 05 49 28](https://github.com/user-attachments/assets/419525cd-1a41-4294-893a-a11056a57fc5)

Also strongly dislike the change to it removing it's air canister and own small power setup as people REALLY like to snip the wire going over to AI and that prevented it from immediately losing power, like it doesn't even have a substation anymore which means you have the AI's turrets tied to sci's/cargo's substation I believe (what the AI chute room and south asteroid maint are powered by). re-adding the power room here would work
![2025-05-17 05 48 48](https://github.com/user-attachments/assets/0ab934eb-6255-41ce-8352-bb28eb0994cc)


Also if you moved east solars why is the catwalk path still going over there? Also there is no catwalk for the door going into AI which looks weird. 

Also why is there like 4 air pipes going through various walls from the outside? the top left one especially looks odd, just have the main door one and maybe the one right off the path to AI and then route the pipes through the rooms as normal.

I'm guessing the chute now ends inside the airlock of AI's asteroid area

Once I get back from my work trip in a few months I'll probably completely rework the AI area myself but this sorta works for now if adjustments are made

OnsenCapy commented:
> Personally really don't like the change to my Leonid having the AI's outer asteroid area being extra fully sealed with walls now, it's not meant to be hard to get to the AI's building.

I compromised and pulled the wall way back, it only goes as far as the other two walls that already existed now. The idea being that if you're entering the AI from the front, you should have to go through the turrets or the walls. But from behind and the sides it being more exposed.

![image](https://github.com/user-attachments/assets/eb429945-6935-4d84-af99-37849ec27316)
![image](https://github.com/user-attachments/assets/f37c216e-b485-4008-ba18-e35c4db5988e)


> Also you placed the blue floor decals on top of the warning stripes making that look weird. place the blue first then the hazard stripes

Fixed.

> Also strongly dislike the change to it removing it's air canister and own small power setup as people REALLY like to snip the wire going over to AI and that prevented it from immediately losing power, like it doesn't even have a substation anymore which means you have the AI's turrets tied to sci's/cargo's substation I believe (what the AI chute room and south asteroid maint are powered by). re-adding the power room here would work 
 
This is something ya missed. The new power area for AI is behind it. I made it with the intention of it being a weak spot to the AI, ideally where nukies would strike on Leonid. It has the SMES and substation, as well as its own (completely secluded from the station) power generation. The air canister

![image](https://github.com/user-attachments/assets/9ece938a-b40b-4793-afea-02acbe1cb7c8)


> Also if you moved east solars why is the catwalk path still going over there? Also there is no catwalk for the door going into AI which looks weird.

The catwalk continues to be used for covering up the cables from AI's backup power area. If you meant the catwalk that leads to the sattelite, i mean, it looks cool, and also still covers MV and LV cables that lead to other areas.

> Also why is there like 4 air pipes going through various walls from the outside? the top left one especially looks odd, just have the main door one and maybe the one right off the path to AI and then route the pipes through the rooms as normal.

I'm still unsure how crossed pipes function. I feel like I heard conflicting information on this all the time. So I didn't have room to run the distro pipes through the sattelite without going on top of the waste line. It shouldn't be an issue regardless, at least not an immediate issue, is what I mean. I hate working with piping and air alarms and all that in mapping...

> I'm guessing the chute now ends inside the airlock of AI's asteroid area

The chute was in a spot where it would throw you directly to the wolves (aka the turrets) if you took it, which is something I changed on Delta as well, since the AI could just turn off the power at the sattelite so people would come, be unable to return, and then get shot to death.

> Once I get back from my work trip in a few months I'll probably completely rework the AI area myself but this sorta works for now if adjustments are made feel free to ignore me if you don't feel like changing stuff again

I don't mind. Leonid is one of the most easy to work with AI sattelites, so it would be difficult to make it weak unless done intentionally.

Gallagin commented:
> The catwalk continues to be used for covering up the cables from AI's backup power area. If you meant the catwalk that leads to the sattelite, i mean, it looks cool, and also still covers MV and LV cables that lead to other areas.


![2025-05-17 22 37 41](https://github.com/user-attachments/assets/e6e8e0d0-9976-44f4-8800-fed47c06930e)
was talking this, adding a single catwalk right there next to the door as a walking path would look nice

OnsenCapy commented:
> was talking this, adding a single catwalk right there next to the door as a walking path would look nice

Yeah I added a catwalk to that too, dw

Moony33 commented:
Oh god, turrets from the space side? I can already see the CE getting shot dead into space and RR'd for going to solars from the outside
